src/main/java/teammates/storage/search/SearchManager.java

```diff
@@ -17,16 +28,52 @@
 
     private static final Logger log = Logger.getLogger();
 
+    private static final String ERROR_DELETE_DOCUMENT =
+            "Failed to delete document(s) %s in Solr. Root cause: %s ";
+    private static final String ERROR_SEARCH_DOCUMENT =
+            "Failed to search for document(s) %s from Solr. Root cause: %s ";
+    private static final String ERROR_SEARCH_NOT_IMPLEMENTED =
+            "Search service is not implemented";
+    private static final String ERROR_PUT_DOCUMENT =
+            "Failed to put document(s) %s into Solr. Root cause: %s ";
+    private static final String STUDENT_COLLECTION_NAME = "students";
+    private static final String INSTRUCTOR_COLLECTION_NAME = "instructors";
+
     private String searchServiceHost;
+    private HttpSolrClient client;
 
     public SearchManager(String searchServiceHost) {
         this.searchServiceHost = searchServiceHost;
+        if (isSearchServiceActive()) {
+            this.client = new HttpSolrClient.Builder(this.searchServiceHost).build();
```

Comment:
If you do this, you should now be able to use existence of `client` to indicate whether search service is active (instead of `searchServiceHost`).